libs/libc/machine/sim/arch_setjmp_arm64.S

```diff
@@ -0,0 +1,108 @@
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <arch/setjmp.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#if defined(CONFIG_HOST_MACOS)
+	#define SYMBOL(x) _##x
+#else
+	#define SYMBOL(x) x
+#endif
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+	.text
+	.globl SYMBOL(setjmp)
+	.align 4
+
+SYMBOL(setjmp):
+
+    stp    x19, x20, [x0, JB_X19_X20]
+    stp    x21, x22, [x0, JB_X21_X22]
+    stp    x23, x24, [x0, JB_X23_X24]
+    stp    x25, x26, [x0, JB_X25_X26]
+    stp    x27, x28, [x0, JB_X27_X28]
+    stp    x29, lr,  [x0, JB_X29_XLR]
+    mov    x1, sp                          /* STP can't access SP directly */
+    stp    fp, x1,   [x0, JB_XFP_XSP]
+    stp    d8, d9,   [x0, JB_D08_D09]
+    stp    d10, d11, [x0, JB_D10_D11]
+    stp    d12, d13, [x0, JB_D12_D13]
+    stp    d14, d15, [x0, JB_D14_D15]
+
+	mov x0,#0						       /* return value */
+	ret
+
+	.text
+	.globl SYMBOL(longjmp)
+	.align 4
```

Comment:
align the code

Reply:
we should use tab in all assemble file like others

Reply:
Okay ... will come with next update.